syntax = "proto3";

message ReadyRequest {
  string newport = 1;
}

message ReadyResponse {
  int32 ready = 1;
}

service ReadyService {
  rpc ServerReady(ReadyRequest) returns (ReadyResponse);
}

message ProcessesRequest {
  //Empty
}

message ProcessesResponse {
  int32 pnumber = 1;
}

message PassChainRequest {
  repeated string chain = 1;
}

message PassChainResponse {
  bool success = 1;
}

service ChainService {
  rpc GetProcesses(ProcessesRequest) returns (ProcessesResponse);
  rpc PassChain(PassChainRequest) returns (PassChainResponse);
}